copier/vcs.py

```diff
@@ -150,29 +126,36 @@ def clone(url: str, ref: OptStr = None) -> str:
     """
 
     location = tempfile.mkdtemp(prefix=f"{__name__}.clone.")
+    _clone = git["clone", "--no-checkout", url, location]
+    # Faster clones if possible
+    if GIT_VERSION >= Version("2.27"):
+        _clone = _clone["--filter=blob:none"]
+    _clone()
 
     if os.path.exists(url) and Path(url).is_dir():
         is_dirty = False
         with local.cwd(url):
             is_dirty = bool(git("status", "--porcelain").strip())
         if is_dirty:
             url_abspath = Path(url).absolute()
-            with TemporaryDirectory(prefix=f"{__name__}.dirty.") as src:
-                with local.cwd(src):
-                    git("--git-dir=.git", f"--work-tree={url_abspath}", "init")
-                    git("--git-dir=.git", f"--work-tree={url_abspath}", "add", "-A")
-                    git(
-                        "--git-dir=.git",
-                        f"--work-tree={url_abspath}",
-                        "commit",
-                        "-m",
-                        "Copier automated commit for draft changes",
-                        "--no-verify",
-                    )
-                    warn(
-                        "Dirty template changes included automatically.",
-                        DirtyLocalWarning,
-                    )
-                    return _perform_clone(src, location, ref)
-
-    return _perform_clone(url, location, ref)
+            with local.cwd(location):
+                # git("--git-dir=.git", f"--work-tree={url_abspath}", "init")
+                git("--git-dir=.git", f"--work-tree={url_abspath}", "add", "-A")
+                git(
+                    "--git-dir=.git",
+                    f"--work-tree={url_abspath}",
+                    "commit",
+                    "-m",
+                    "Copier automated commit for draft changes",
+                    "--no-verify",
+                )
+                warn(
+                    "Dirty template changes included automatically.",
+                    DirtyLocalWarning,
+                )
+
+    with local.cwd(location):
+        git("checkout", ref or "HEAD")
+        git("submodule", "update", "--checkout", "--init", "--recursive", "--force")
```

Comment:
I've got the feeling that this block should be done before importing the ditty changes. However, if the test goes green I guess I'm just wrong.

Reply:
I think it shouldn't matter if it runs before or after. But if there is no HEAD in the case of a recently intialized repo, then this fails and would need to be called after as well. I also realized that specifying `ref` means that dirty changes won't be considered and so added that to the dirty checking logic.